### PR TITLE
Add support for custom tooltip in line

### PR DESCRIPTION
**Pull request by jbenua**

### CODE REVIEW

File: packages/nivo-line/stories/line.stories.js [hunk 1]
```diff
@@ -271,3 +271,33 @@ stories.add('with formatted values', () => (
...
+    <Line
+        {...commonProperties}
+        enableStackTooltip={true}
+        curve={select('curve', curveOptions, 'linear')}
+        tooltip={slice => {
+            return (
+                <div>
+                    <h2>{slice.id}</h2>
+                    {slice.points.map((e, i) => {
+                        console.log(e)
```
Comment (Panczo) [inline packages/nivo-line/stories/line.stories.js]: You can remove console.log

**Comment by plouc** [comment]
@jbenua sorry, don't have much time for side projects immediately, but I'll try to have a look at this asap

**Comment by Panczo** [comment]
@jbenua For me it's a great work. Thanks :)

**Comment by taugustyn** [comment]
Please merge and release an update on NPM. Greatly appreciated! <3

**Comment by janisknets** [comment]
We have a use case which would benefit from this change, waiting on the merge

**Comment by plouc** [comment]
@jbenua, thank you for your contribution! and sorry for the delay :)

**Comment by taugustyn** [comment]
Thank you so much! :D

File: packages/nivo-line/src/LineSlices.js [hunk 2]
```diff
@@ -45,6 +54,8 @@ LineSlices.propTypes = {
     showTooltip: PropTypes.func.isRequired,
     hideTooltip: PropTypes.func.isRequired,
     theme: PropTypes.object.isRequired,
+    tooltip: PropTypes.element,
```
Comment (taugustyn) [inline packages/nivo-line/src/LineSlices.js]: @jbenua Should it be either `element` or `func`? I am getting errors when I pass `tooltip` like this ```
tooltip={
              slice => {
                return (
                  <div>
                    <h2>{slice.id}</h2>
                    {slice.points.map((e, i) => (
                      <p key={i}>
                        <strong>{e.id}:</strong> {e.value}
                      </p>
                    ))}
                  </div>
                )
              }
            }
```

File: packages/nivo-line/src/LineSlicesItem.js [hunk 3]
```diff
@@ -54,34 +54,44 @@ LineSlicesItem.propTypes = {
     hideTooltip: PropTypes.func.isRequired,
     isHover: PropTypes.bool.isRequired,
     theme: PropTypes.object.isRequired,
+    tooltip: PropTypes.element,
```
Comment (taugustyn) [inline packages/nivo-line/src/LineSlicesItem.js]: Same here @jbenua, tooltip should also allow `PropTypes.func`, it even checks for it in line 84:
```
renderContent={
  typeof tooltip === 'function' ? tooltip.bind(null, { ...slice }) : null
}
```

Comment (plouc) [reply packages/nivo-line/src/LineSlicesItem.js]: I've fixed this in #275.

Comment (taugustyn) [reply packages/nivo-line/src/LineSlicesItem.js]: Thanks Raphaël, I will watch master/npm when its merged.

**Comment by santhoshCharles** [comment]
The scroll is not working in Custom Tooltip for Line Charts. can anyone help me on this?

**Pull request merged**